src/cloudWatchLogs/activation.ts

```diff
@@ -39,11 +39,21 @@ export async function activate(context: vscode.ExtensionContext, configuration:
     context.subscriptions.push(
         vscode.workspace.onDidCloseTextDocument(doc => {
             if (doc.isClosed && doc.uri.scheme === CLOUDWATCH_LOGS_SCHEME) {
+                registry.disposeRegistryData(doc.uri)
                 registry.deregisterLog(doc.uri)
             }
         })
     )
 
+    context.subscriptions.push(
+        vscode.workspace.onDidChangeTextDocument((event: vscode.TextDocumentChangeEvent) => {
+            const eventUri = event.document.uri
+            if (registry.hasLog(eventUri) && !isLogStreamUri(eventUri)) {
+                registry.highlightDocument(eventUri)
```

Comment:
onDidChangeTextDocument is potentially very frequent. Doesn't matter here in practice because the doc is readonly, but it's a hidden spec, which is debt (a potential future bug). What's the reason for calling highlightDocument() on this event instead of just doing it explicitly after we populate the text?

Reply:
Before having this be an event handler I tried having `highlightDocument` in both the `updateLog` function and the `documentProvider.provideTextDocumentContent` function. It seems that both fail to highlight newly loaded data from the `load more` codelense before it is rendered to the screen. I couldn't figure out where else to have to other than on a document change so that this behavior worked as expected.

Reply:
> just doing it explicitly after we populate the text?

Because we don't populate the text document. VS Code does.

Reply:
Ok. That kind of hard-gained knowledge is worth a 1-line comment here. Something like

    // Do highlight on text-changed because we don't control when text is populated by vscode, also for "load more" codelens.